### ISSUE TITLE
Getting  { [Error: getaddrinfo ENOTFOUND] code: 'ENOTFOUND', errno: 'ENOTFOUND', syscall: 'getaddrinfo' }

### ISSUE THREAD

**Issue by sebwaks** [Azure/azure-sdk-for-node#1295]
Node.js 0.10.31 and azure sdk 0.9.15 and with a bit more heavy load when using servicebus I'm getting 
{ [Error: getaddrinfo ENOTFOUND] code: 'ENOTFOUND', errno: 'ENOTFOUND', syscall: 'getaddrinfo' }.

**Comment by sescandell** [comment]
Hi there,

Having the same issue as the initial post here.
I'm running:
 - Node v0.12.2
 - Azure SDK NodeJS v0.10.5

I'm trying to use serviceBus. But when, executing any request, I'm getting the following error:
> {"code":"ENOTFOUND","errno":"ENOTFOUND","syscall":"getaddrinfo","hostname":"mynamespacehere-sb.accesscontrol.windows.net"}

Here is the code used:
```js
    // ... Env variables: AZURE_SERVICEBUS_NAMESPACE & AZURE_SERVICEBUS_ACCESS_KEY
    azureServiceBusReceiver = azure.createServiceBusService();
    azureServiceBusReceiver.listTopics(function(results){
	console.log(process.pid + " - " + 'listTopics: ' + JSON.stringify(arguments));
    });
```

Any idea?

Many htanks,

**Comment by farstar** [comment]
redefine AZURE_STORAGE_DNS_SUFFIX environment variable fixed my issue.

**Comment by mauricionr** [comment]
Hi there,
Having the same issue as the initial post here.
```
serviceBusClient.getTopic(topic, function (error, top1) {
serviceBusClient.createTopic(topic, function (error, top) {
serviceBusClient.getSubscription(topic, subscription, function (error2, sub) {
serviceBusClient.createSubscription(topic, subscription, function (error3, sub) {
```

Service Bus error: {"code":"ENOTFOUND","errno":"ENOTFOUND","syscall":"getaddrinfo","hostname":"socialcontrol.servicebus.windows.net"}

**Comment by Tomtomgo** [comment]
I'm getting the same issue. Fixed it via: https://datapsyche.wordpress.com/2015/07/01/azure-service-bus-topic-issue-while-using-nodejs-sdk-error-getaddrinfo-enotfound/.

**Comment by renanmpn** [comment]
Hi there,

I'm having a simular problem.

After I upload like a thousand files to a blob, my system stop to send more file. 
Another method that verifies how many messages are in the queue stops too.

Here is the error:

{ [Error: getaddrinfo EADDRINFO] code: 'EADDRINFO', errno: 'EADDRINFO', syscall: 'getaddrinfo' }


Any hint?

**Comment by jeffwilcox** [comment]
@mayurid is there anyone looking into fixing this? Sounds like some invalid URL concatenation that's been present for a very, very long time.

@Tomtomgo seems to have offered a good workaround in the meantime.

**Comment by fiveisprime** [comment]
This is one of two things:

Your application is sending enough messages that you are running out of file descriptors (unlikely) or you are seeing a bug in libuv because of the version of node that's being used.

If bumping the descriptor limit doesn't help (it won't since these instances run on Windows), then you'll need to upgrade to a later version of node. This issue was fixed in libuv 0.10 and eventually landed in node@0.10 later on. I don't remember the exact version, but either way, you should be on stable over a 0.x version if possible.

It's also likely that this thread died out because the default node version changed to a later version and nobody has seen this since. In that case, it can safely be closed. :)

(reference issue from node@0.x: https://github.com/nodejs/node-v0.x-archive/issues/7742)

**Comment by jtaubensee** [comment]
Closing per comment from @fiveisprime
